File: .LABS/l1.java
import java.util.Scanner;

public class l1{

    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);
        int a, b, c;

        System.out.println("a:");
        a = in.nextInt();

        System.out.println("b:");
        b = in.nextInt();

        System.out.println("c:");
        c = in.nextInt();

        int maxN = Math.max(Math.max(a, b), c);
        int minN = Math.min(Math.min(a, b), c);
        int midN = (a + b + c) - (minN + maxN);

        int difmin = minN - midN;
        int difmax = maxN - midN;

        double bigdif = Math.max(Math.abs(difmin), Math.abs(difmax));
        double sign = Math.copySign(1, difmin+difmax);
        double delta = sign*bigdif;

        double result = midN+delta;

        System.out.println(result);
        in.close();
    }
}

File: .LABS/l1.py
from email.mime.nonmultipart import MIMENonMultipart
from math import copysign


a = input()
b = input()
c = input()

maxnum = max(a, b, c)
minnum = min(a, b, c)
midnum = (a + b + c) - (minnum + maxnum)

difmin = minnum - midnum
difmax = maxnum - midnum

bigdif = max(abs(difmin), abs(difmax))
sign = copysign(1, difmin+difmax)
delta = sign*bigdif

result = midnum + delta

print(result)